Feat/client env file

vikvikvr opened the pull request:
## improve .env file usage

Server was not getting `DB_NAME`  from .env file to create connection string when used in production.

## server: add dev start script

Added script based on `nodemon` for live-realod.

Execute `npm run start:dev` to use it.

## server: add prettier config

Prettier config was missing for the server.

I have set `singleQuote` to false (while on the client is true) to minimize the changes to the existing code.

## weather: improve error handling

Now every status code except for `200` is considered an error, `401` used to break the app.

This emerged by creating a fresh open-weather API key, which requires up to 1 hour to be fully active.

see https://openweathermap.org/faq

GILREICH1 commented on server/models/database.js:
```diff
@@ -1,21 +1,28 @@
+    return `mongodb://localhost:27017/${DB_NAME}`;
+  }
+
+  return `mongodb+srv://${DB_USER}:${DB_USER_PASSWORD}@cluster0.9g4e8.mongodb.net/${DB_NAME}?retryWrites=true&w=majority`;
```
will using a different URI (uses '9g4e8' vs 'kbmrd') not prevent connection the currently used database which holds all the tasks? Maybe this part of the URI should also be dynamic?

vikvikvr replied:
You are absolutely right, I forgot and commited that part as well.

It should come from the .env as well, maybe as `DB_CLUSTER` or something

vikvikvr replied:
I've added `DB_HOST` to the `.env` file.

Remember to update environment settings on **Heroku** when it's deployed.

Also: moved **Prettier** config to the root folder so it's shared between client and server 😄

GILREICH1 reviewed (changes requested):
nice! much cleaner 👍

vikvikvr commented:
I've added `DB_HOST` to the `.env` file.

Remember to update environment settings on **Heroku** when it's deployed.

Also: moved **Prettier** config to the root folder so it's shared between client and server 😄
